src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -221,9 +221,19 @@ private void handleRequest(ChannelHandlerContext ctx, FullHttpRequest req) throw
                 return;
             }
 
+            final String path = stripQuery(req.uri());
+            // Reject the requests with non-absolute path except the 'OPTIONS *' request.
+            if (path.isEmpty() || path.charAt(0) != '/') {
+                if (req.method() == HttpMethod.OPTIONS && "*".equals(path)) {
+                    respond(ctx, reqSeq, req, HttpResponseStatus.NOT_IMPLEMENTED);
```

Comment:
To implement properly, we need to add an `Allow` header to the response. It's not a lot of work, so I just did it and changed the response code to 200.